### PR TITLE
v1.10 backports 2022-06-07

### PR DESCRIPTION
**Pull request by joamaki**
* #19347 -- Improve endpoint and DNS proxy lock contention during bursty DNS traffic (@christarazi)
 * #19570 -- pkg/policy/api: Optimize FQDNSelector String() (@christarazi)
 * #19423 -- pkg/labels: Optimize SortedList() and FormatForKVStore() (@christarazi)
 * #19383 -- daemon, fqdn: Add flag to control FQDN regex LRU size (@christarazi)
 * #19857 -- Expose metrics for active FQDN connections per endpoint (@christarazi)
 * #19987 -- metrics: Fix NaN value for cilium metrics list CLI (@sayboras)
 * #19933 -- helm: use port 80/443 by default for the peer service (@rolinh)
 * #19925 -- pkg/fqdn: Fix missing delete for forward map (@christarazi)
 * #19959 -- clustermesh: Add ownerReferences for CiliumNodes (@sayboras)
 * #19999 -- Add type label to the identity metric (@ungureanuvladvictor)
 * #20011 -- bugtool: Add structured node and health output (@gandro)

PRs skipped due conflicts:

 * #19565 -- ui: v0.9.0 images and drop envoy proxy container (@geakstr)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 19347 19570 19423 19383 19857 19987 19933 19925 19959 19999 20011; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label BRANCH=v1.10 ISSUES=19347,19570,19423,19383,19857,19987,19933,19925,19959,19999,20011
```

### CODE REVIEW

File: .github/workflows/tests-smoke.yaml [hunk 1]
```diff
@@ -180,8 +180,14 @@ jobs:
...
-          GO111MODULE=on go get github.com/prometheus/prometheus/cmd/promtool@e4487274853c587717006eeda8804e597d120340 # This is commit hash for v2.24.1
-          cat metrics.prom | $HOME/go/bin/promtool check metrics
+          # Install promtool binary release. `go install` doesn't work due to
+          # https://github.com/prometheus/prometheus/issues/8852 and related issues.
+          curl -sSL --remote-name-all https://github.com/prometheus/prometheus/releases/download/v${PROM_VERSION}/{prometheus-${PROM_VERSION}.linux-amd64.tar.gz,sha256sums.txt}
+          sha256sum --check --ignore-missing sha256sums.txt
+          tar xzvf prometheus-${PROM_VERSION}.linux-amd64.tar.gz prometheus-${PROM_VERSION}.linux-amd64/promtool
+          rm -f prometheus-${PROM_VERSION}.linux-amd64.tar.gz
+          sudo mv prometheus-${PROM_VERSION}.linux-amd64/promtool /usr/bin
+          cat metrics.prom | promtool check metrics
```
Comment (sayboras) [inline .github/workflows/tests-smoke.yaml]: is this change required ? Basically, I can't find where PROM_VERSION is defined :thinking:

**Review by sayboras — COMMENTED** [review]
Small comment on GHA, the rest looks good for my changes.

File: go.mod [hunk 2]
```diff
@@ -121,6 +121,9 @@ replace (
 
 	go.universe.tf/metallb => github.com/cilium/metallb v0.1.1-0.20210607221240-b4c60b959dd7
 
+	// Make sure to use grpc v1.29.1 as grpc/naming package was removed in newer versions.
+	google.golang.org/grpc => google.golang.org/grpc v1.29.1
```
Comment (joamaki) [inline go.mod:125]: This was added due to:
```
go: finding module for package google.golang.org/grpc/naming
github.com/cilium/cilium/pkg/kvstore imports
	go.etcd.io/etcd/clientv3 tested by
	go.etcd.io/etcd/clientv3.test imports
	go.etcd.io/etcd/integration imports
	go.etcd.io/etcd/proxy/grpcproxy imports
	google.golang.org/grpc/naming: module google.golang.org/grpc@latest found (v1.47.0), but does not contain package google.golang.org/grpc/naming
```

Perhaps better would be to bump etcd here as well? Thoughts?

File: .github/workflows/tests-smoke.yaml (hunk 1 above)
Comment (joamaki) [reply .github/workflows/tests-smoke.yaml]: Discussed with @sayboras. Dropping this commit.

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my changes :medal_military:

**Comment by joamaki** [comment]
/test-backport-1.10

**Review by gandro — APPROVED** [review]
My commit looks good

**Review by rolinh — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]
`github-sec` review is not required anymore following drop of commit editing the workflow files, approving just to get rid of the review request.

**Review by michi-covalent — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (christarazi) [reply go.mod:125]: Probably fine to keep the grpc lib pinned here. This is a v1.10 branch which we won't typically expect changes to vendor moving forward.

**Review by christarazi — APPROVED** [review]
Thanks, my commits look good!

**Pull request merged**